### ISSUE TITLE
Doesn't sort when props.data changes

### ISSUE THREAD

**Issue by ishmulyan** [react-bootstrap-table/react-bootstrap-table2#210]
Let's say we have Container which renders table of products which are updating with some interval.

```
class Container extends React.Component {
  constructor(props) {
    super(props);

    this.state = { products: productsGenerator() };

    //async update of products
    setInterval(() => this.setState({ products: productsGenerator() }), 1000);
  }

  render() {
    return (
      <div>
        <BootstrapTable keyField="id" data={ this.state.products } columns={ columns } />
        <Code>{ sourceCode }</Code>
      </div>
    );
  }
}

export default Container;
```

Since BootstrapTable uses props.data for passing sorted data instead of the state it makes the case described above not possible to implement.
Is this intended behavior?

**Comment by AllenFang** [comment]
HI @ishmulyan, it's sorry that I cant answer you without some confirmation:

1. did you enable `remote` mode?

> Since BootstrapTable uses props.data for passing sorted data

I can understand this..... u can pass unsort data actually..

> it makes the case described above not possible to implement.

What case, please describe your case more details

Thanks!

**Comment by ishmulyan** [comment]
Hi @AllenFang!

1. I didn't. As far as I understand `remote` is for custom sort.
2. Clicking on header sorts data and calls `componenWillReceiveProps` with sorted data, so if I click on some header it sorts data, but when my data is changing it propagates to BootstrapTable and it renders unsorted data.

Thanks!

**Comment by AllenFang** [comment]
@ishmulyan sorry for lately reply, 

> I didn't. As far as I understand remote is for custom sort.
Actually, the remote mode allow you to handle your data logic(ex: sort, filter, paging etc) external, which mean `react-bootstrap-table2` will not handle the data instead you application need to handle.

 ok, I understand you problem, actually, this is a unimplement feature.

in legacy `react-bootstrap-table`, we allow developer to manage the sorting, so in this case, if you dont enable the remote mode, you are supposed to manage your sorting state, but unfortunately, `react-bootstrap-table2` doesn't support this feature, I'll fix it ASAP

BTW, I can also consider other solution is that keep the sorting state without external management, let me figure it firstly. also, need some time on this issue, I'm busy recently.

let me know if you got any questions, thanks !!!

**Comment by AllenFang** [comment]
Fixed on `react-bootstrap-table-next@0.1.4`, check [release note](https://react-bootstrap-table.github.io/react-bootstrap-table2/blog/2018/03/06/version-bump.html), thank!
